File: frontend/src/api/order.service.js
import api from './axios';

export const orderService = {
    createOrder: async (orderData, email) => {
        const response = await api.post(`/order-service/api/orders?email=${email}`, orderData);
        return response.data;
    },

    getOrderById: async (id) => {
        const response = await api.get(`/order-service/api/orders/${id}`);
        return response.data;
    },

    getMyOrders: async (userId) => {
        const response = await api.get(`/order-service/api/orders/user/${userId}`);
        return response.data;
    }
};

File: Backend/services/order-service/src/main/java/com/shopverse/orderService/service/OrderService.java
package com.shopverse.orderService.service;


import com.shopverse.orderService.entity.Order;



import java.util.List;

public interface OrderService {

    Order createOrder(Long userId, String productName, Integer quantity, Double totalAmount, String email, List<com.shopverse.orderService.dto.OrderItemDto> items);

    Order getOrderById(Long id);

    List<Order> getAllOrders();

    Order cancelOrder(Long id);

    String initiatePayment(Long orderId, String email, String token);

    List<Order> getOrdersByUserId(Long userId);

    void updateOrderStatus(Long orderId, String status);
}
